internal/infrastructure/storage/mem.go

```diff
@@ -0,0 +1,21 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/mihailtudos/metrickit/internal/domain/entities"
+)
+
+type MemStorage struct {
+	Counter map[string]entities.Counter
+	Gauge   map[string]entities.Gauge
+	Mu      sync.Mutex
```

Comment:
It shouldn't be exported. You just need to add methods to this structure that allwow working with the two maps.